### DISCUSSION TITLE
How to manipulate an agent's specific attribute based on a condition in the model?

### DISCUSSION THREAD

**Discussion by ghost** [mesa/mesa#1681] [Q&A] [answered]
I have created senders and carriers as two classes of agents. They have unique IDs. However, I want to change other attributes of a sender or a carrier using the unique id. Is there a way to create object names so that I can use them whenever needed to gain access to the object-specific attributes? I have tried the following but it throws an error: 'str' object has no attribute 'unique_id'

<img width="784" alt="image" src="https://user-images.githubusercontent.com/131771458/235462472-fcfeaf8c-d18e-49ca-9ae2-ae103801265d.png">

**Comment by rht** [comment]
It depends on the scheduler class you are using. With `RandomActivationByType`, you can access an agent by `self.schedule.agents_by_type[agent_class][unique_id_str]`. See https://github.com/projectmesa/mesa/blob/dc4d5fcfe5f2e5d25e20dcf5eebe9e1f5aee80e3/mesa/time.py#L244

**Reply by ghost** [reply]
In the following code, will I actually need the object name to refer to its Unique_id? 

self.schedule.agents_by_type[agent_class][unique_id]

When I tried to use this in the model, it says it is unable to locate "unique_id". I thought of using "self.Unique_id" but this will work only if I am writing this in the agent's class and not outside the class (Model).

**Reply by ghost** [reply, marked as answer]
I was able to create a dictionary with the key as the unique id of the agents and the value as the object name which looks like this:

**_<Abstract_crowdlogistics_model.Senders object at 0x7fd7dabe39d0>_**.

Now I am able to access Sender's attribute in a function that I defined in the Carrier agent class as shown below:

for i in self.model.Sender_population.keys():
                if (self.model.Sender_population[i].Participation_status == "Y" and len(self.model.Sender_population[i].pickup_request_list)!= 0):
                    self.Potential_senders_to_match.append(i)
